src/lib/pci_virtio_sock.c

```diff
@@ -1020,12 +1022,13 @@ static void send_response_common(struct pci_vtsock_softc *sc,
 }
 
 static void send_response_sock(struct pci_vtsock_softc *sc,
-				 uint16_t op, uint32_t flags,
-				 const struct pci_vtsock_sock *sock)
+			       uint16_t op, uint32_t flags,
+			       struct pci_vtsock_sock *sock)
 {
 	send_response_common(sc, sock->local_addr, sock->peer_addr,
 			     op, VIRTIO_VSOCK_TYPE_STREAM, flags,
 			     sock->buf_alloc, sock->fwd_cnt);
+	sock->credit_cnt = sock->fwd_cnt;
```

Comment:
No, this is not the correct place. I was a bit uncertain about this one but I agree with your assessment that it is premature to update `sock->credit_cnt` here.